test/test_overrides.py

```diff
@@ -1506,6 +1504,51 @@ class A(torch.Tensor):
             finally:
                 del g
 
+        counter = [0]
+
+        class Foo:
```

Comment:
Are these changes enough for you to add an E2E test of a subclass that has both a torch_function and torch_dispatch impl, that we can run torch.compile on? (that might also be a useful test to add in this PR)

Reply:
Ah do you mean to make sure that we can now use this disable-non-infra torch function subclass context manager to ensure that torch function is disabled? This is currently tested in the next PR, but I guess I can move it to this one.

Reply:
np if you have a similar test further in the stack